library/common.h

```diff
@@ -334,4 +334,25 @@ static inline void mbedtls_xor_no_simd(unsigned char *r,
 #define MBEDTLS_OPTIMIZE_FOR_PERFORMANCE
 #endif
 
+/* Suppress compiler warnings for unused functions and variables. */
+#if !defined(MBEDTLS_MAYBE_UNUSED) && defined(__GNUC__)
+#define MBEDTLS_MAYBE_UNUSED __attribute__((unused))
+#endif
+#if !defined(MBEDTLS_MAYBE_UNUSED) && defined(__IAR_SYSTEMS_ICC__) && defined(__VER__)
+#if (__VER__ >= 8010000) // IAR 8.1 or later
+#define MBEDTLS_MAYBE_UNUSED __attribute__((unused))
+#endif
+#endif
+#if !defined(MBEDTLS_MAYBE_UNUSED) && defined(_MSC_VER)
```

Comment:
IIRC when clang is used as the compiler driver by Visual Studio, it doesn't define `__GNUC__` and does define `_MSC_VER`, but only understands the GCC-style attributes, so I think we need

```
#if !defined(MBEDTLS_MAYBE_UNUSED) && (defined(__GNUC__) || defined(__clang__)
```

I did a quick'n'dirty install of this in a VM for testing some weeks ago - I'll see if I can dig it up and test again

Reply:
I suspect a lot of the codebase will get this wrong, but I'll fix this instance.

Reply:
Would moving the `defined(__has_attribute)` up take care of that case? (Even with that, we do want the `__GNUC__` case, at least, for older GCC versions that predate `__has_attribute`.)